### ISSUE TITLE
Check `.hasAttribute()` in `prefer-dom-node-dataset` rule

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1669]
[MDN suggest using `in` operator to check existence](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#accessing_values), I'm not sure what's the best, maybe `hasOwnProperty()`?

## Fail

```js
element.hasAttribute('data-foo');
```

## Pass

```js
"foo" in element.dataset;
```

**Comment by sindresorhus** [comment]
Yes, it should recommend `.hasOwnProperty()`. We can add a code TODO comment to switch to `Object.hasOwn` later on.
